opensrp-giz-malawi/src/test/java/org/smartregister/giz/processor/GizMalawiProcessorForJavaTest.java

```diff
@@ -0,0 +1,216 @@
+package org.smartregister.giz.processor;
+
+import android.content.ContentValues;
+import android.content.Context;
+
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+import org.smartregister.child.util.Utils;
+import org.smartregister.domain.db.Client;
+import org.smartregister.domain.db.Event;
+import org.smartregister.domain.db.EventClient;
+import org.smartregister.domain.jsonmapping.Table;
+import org.smartregister.giz.activity.ChildImmunizationActivity;
+import org.smartregister.giz.application.GizMalawiApplication;
+import org.smartregister.growthmonitoring.domain.Height;
+import org.smartregister.growthmonitoring.repository.HeightRepository;
+import org.smartregister.immunization.domain.ServiceRecord;
+import org.smartregister.immunization.domain.ServiceSchedule;
+import org.smartregister.immunization.domain.ServiceType;
+import org.smartregister.immunization.domain.VaccineSchedule;
+import org.smartregister.immunization.repository.RecurringServiceRecordRepository;
+import org.smartregister.immunization.repository.RecurringServiceTypeRepository;
+import org.smartregister.repository.DetailsRepository;
+
+import java.util.Arrays;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({GizMalawiApplication.class, Utils.class, VaccineSchedule.class, ServiceSchedule.class})
+public class GizMalawiProcessorForJavaTest {
+
+    @Mock
+    private GizMalawiApplication gizMalawiApplication;
+
+    @Mock
+    private org.smartregister.Context openSrpContext;
+
+    @Mock
+    private HeightRepository heightRepository;
+
+    @Mock
+    private ContentValues contentValues;
+
+    @Captor
+    private ArgumentCaptor<Height> processHeightArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<ServiceRecord> recordServiceArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor addDetailsRepoArgumentCaptor;
+
+    @Mock
+    private RecurringServiceTypeRepository recurringServiceTypeRepository;
+
+    @Mock
+    private RecurringServiceRecordRepository recurringServiceRecordRepository;
+
+    private GizMalawiProcessorForJava processorForJava;
+
+    @Mock
+    private DetailsRepository detailsRepository;
+
+    @Before
+    public void setUp() {
+        processorForJava = Mockito.spy(GizMalawiProcessorForJava.getInstance(Mockito.mock(Context.class)));
+    }
+
+    @Test
+    public void processHeightWithEventClientNull() throws Exception {
```

Comment:
Better naming of test, I am having to read the code in the tests to figure out what you wanted to achieve